File: app/src/main/java/com/gee12/mytetroid/RecordFieldsSelector.java
package com.gee12.mytetroid;

import android.content.Context;

import java.util.Set;

/**
 * Класс для заполнения переменной-маски для быстрого получения выбранных значений
 *  из множественного списка.
 */
public class RecordFieldsSelector extends StringsIntMask {

    private static final int AUTHOR_INDEX = 0;
    private static final int TAGS_INDEX = 1;
    private static final int CREATED_INDEX = 2;
    private static final int EDITED_INDEX = 3;

    /**
     *
     * @param context
     * @param option Список выбранных значений.
     */
    public RecordFieldsSelector(Context context, Set<String> option) {
        super(option, context.getResources().getStringArray(
                (App.INSTANCE.isFullVersion())
                    ? R.array.record_fields_in_list_entries_pro
                    : R.array.record_fields_in_list_entries));
    }

    public boolean checkIsAuthor() {
        return checkValue(AUTHOR_INDEX);
    }

    public boolean checkIsTags() {
        return checkValue(TAGS_INDEX);
    }

    public boolean checkIsCreatedDate() {
        return checkValue(CREATED_INDEX);
    }

    public boolean checkIsEditedDate() {
        return checkValue(EDITED_INDEX);
    }
}

File: app/src/main/java/com/gee12/mytetroid/RecordFieldsSelector.kt
package com.gee12.mytetroid

import android.content.Context
import com.gee12.mytetroid.helpers.AppBuildHelper

/**
 * Класс для заполнения переменной-маски для быстрого получения выбранных значений
 * из множественного списка.
 *
 * @param context
 * @param option Список выбранных значений.
 */
class RecordFieldsSelector(
    context: Context,
    appBuildHelper: AppBuildHelper,
    option: Set<String?>?
) : StringsIntMask(
    option,
    context.resources.getStringArray(
        if (appBuildHelper.isFullVersion()) R.array.record_fields_in_list_entries_pro else R.array.record_fields_in_list_entries
    )
) {
    companion object {
        private const val AUTHOR_INDEX = 0
        private const val TAGS_INDEX = 1
        private const val CREATED_INDEX = 2
        private const val EDITED_INDEX = 3
    }

    fun checkIsAuthor(): Boolean {
        return checkValue(AUTHOR_INDEX)
    }

    fun checkIsTags(): Boolean {
        return checkValue(TAGS_INDEX)
    }

    fun checkIsCreatedDate(): Boolean {
        return checkValue(CREATED_INDEX)
    }

    fun checkIsEditedDate(): Boolean {
        return checkValue(EDITED_INDEX)
    }

}